### ISSUE TITLE
cargo install fails under Windows

### ISSUE THREAD

**Issue by msuesskraut** [drahnr/cargo-spellcheck#14]
Installing cargo-spellcheck fails under Windows 10:

> cargo install cargo-spellcheck
>    Updating crates.io index
>  Installing cargo-spellcheck v0.1.0
>   ...
>   Compiling hunspell-sys v0.1.3
> error: failed to run custom build command for `hunspell-sys v0.1.3`
>
> Caused by:
>  process didn't exit successfully: `C:\Users\<...>\AppData\Local\Temp\cargo-install3YNVZA\release\build\hunspell-sys-9a099962c3dfc468\build-script-build` (exit code: 101)
> --- stdout
> running: "autoreconf" "-ivf"
> 
> --- stderr
> thread 'main' panicked at '
> failed to execute command: Das System kann die angegebene Datei nicht finden. (os error 2)
> is `autoreconf` not installed?
> 
> build script failed, must exit now', C:\Users\<...>\.cargo\registry\src\github.com-1ecc6299db9ec823\autotools-0.2.1\src\lib.rs:449:5
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> 
> warning: build failed, waiting for other jobs to finish...
> error: failed to compile `cargo-spellcheck v0.1.0`, intermediate artifacts can be found at `C:\Users\<...>\AppData\Local\Temp\cargo-install3YNVZA`
>
> Caused by:
>   build failed

I have installed `hunspell` with chocolatey. But it does not help. Chocolatey does not install any libs and it seams that hunspell-sys contains the hunspell code and tries to build it with the autotools, which just are not available under Windows.

I recognize, that the error is not in `cargo-spellcheck` itself but in its dependencies. Has someone expirence to install `cargo-spellcheck` under Windows? If yes, it would be nice to document, how to do this.

**Comment by drahnr** [comment]
https://github.com/euclio/hunspell-sys/blob/master/build.rs#L18-L22 shows that if there is no `hunspell` library picked up by package-config, it tries to build one with autotools - so either you make `pkg-config` aware of your installation, or you install autotools too (caveat: I have very limited experience with windows)

**Comment by drahnr** [comment]
It would be awesome if you could provide some more detail or if you already made it work, provide the steps to get there so I can add it to the README file :) thanks!

**Comment by msuesskraut** [comment]
I tried to make it work yesterday. The issue is, as far as I know, that
hunspell itself does only compile under Windows with a Unix-like Setup
(either Msys2 or cygwin). Whereas rust itself is perfectly usable with an
out of the box Windows.

I will try to compile cargo-spellcheck in msys2 and will test it. I will
report back how it works. However, next week I don't have time for that -
hence it will take a while.

**Comment by drahnr** [comment]
I could potentially provide binaries as part of the releases, which would contain cross compiled windows binaries - fedora provides mingw64 precompiled hunspell libs, but that really would be more of a band-aid solution.

**Comment by msuesskraut** [comment]
What I found out: `hunspell-sys` does not compile with the msvc build chain. `hunspell` itself contains a msvc project, but that failed with Visual Studio 2017 for me.

However, it is possible to compile `hunspell` with `msvc` in the `build.rs` of `hunspell-sys`. Tests of `hunspell-sys` pass.

I submitted a pull-request to `hunspell-sys`: https://github.com/euclio/hunspell-sys/pull/2 with that hopefully `hunspell-sys` becomes also usable under Windows with `x86_64-pc-windows-msvc` rust. And, hence, `cargo-spellcheck` should become installable by `cargo install cargo-spellcheck`.

Btw. using `msys2` with my `msvc` based rust installation did not work.
